instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/yaml/MetricStructure.java

```diff
@@ -51,6 +59,36 @@ public void setUnit(String unit) {
     this.unit = validateUnit(unit.trim());
   }
 
+  void setStateMapping(Map<String, Object> stateMapping) {
```

Comment:
I think empty state mapping shouldn't be allowed, as it makes no sense, even if does not seem to cause any harm.